django/db/backends/sqlite3/operations.py

```diff
@@ -282,15 +282,13 @@ def convert_datetimefield_value(self, value, expression, connection):
         return value
 
     def convert_datefield_value(self, value, expression, connection):
-        if value is not None:
-            if not isinstance(value, datetime.date):
-                value = parse_date(value)
+        if value is not None and not isinstance(value, datetime.date):
+            value = parse_date(value)
         return value
 
     def convert_timefield_value(self, value, expression, connection):
-        if value is not None:
-            if not isinstance(value, datetime.time):
-                value = parse_time(value)
+        if value is not None and not isinstance(value, datetime.time):
+            value = parse_time(value)
```

Comment:
Please revert changes in converters, these conditions are prepare for handling different types of values. I don't see any value in changing this.